tests/backend/test_fock_measurement.py

```diff
@@ -156,3 +156,21 @@ def test_vacuum_measurements(self, setup_backend, pure):
 
             meas = backend.measure_fock([0, 1, 2])[0]
             assert np.all(np.array(meas) == 0)
+
+
+    def test_coherent_state_has_photons(self, setup_backend, pure):
+        """Test that a coherent state with a mean photon number of 4 and sampled NUM_REPEATS times will produce photons"""
+        backend = setup_backend(1)
+        alpha = 2.0
+        nonzero = False
+
+        for _ in range(NUM_REPEATS):
+            backend.reset(pure=pure)
+            backend.displacement(alpha, 0)
+            meas = backend.measure_fock([0])[0]
+            if meas > 0:
+                nonzero = True
+                break
+        assert nonzero
```

Comment:
@josh146 @nquesada any thoughts on how we might test the same thing (or something similar enough) deterministically?

Reply:
The test looks stochastic and ugly (I am not a fan of that break but it was the simpler than doing a while with two conditionals).
The probability of getting vacuum for this particular a state with \alpha = 2 is \exp(-4) ~ 0.018.
We are checking this NUM_REPEATS = 50 times so the chance that in those 50 trials we don't get at least a nonzero event is 10^-87. This test is in practice deterministic.